### ISSUE TITLE
Remotely connecting to jupyter in PyCharm

### ISSUE THREAD

**Issue by PiotrowskiD** [jupyter/jupyter#294]
Unfortunately remote interpreter for jupyter notebook in PyCharm is for now not supported but is there a way around that? For example using jupyterhub or somehow binding normal jupyter to one port set auth to password instead of token and setting up ssh tunneling for that port or something like that? I could connect to server working remotely using jupyterhub and tunneling but only through the brower as in the PyCharm I got 405 POST. I don't have access to admin myself but I can ask for some configuration. So the solution can require some configuration to be done but it has to be done once not every time I would like to use it. I will also need python 2.7 kernel there but I hope this -> https://github.com/jupyter/jupyter/issues/71 will be enough to do that.
 
TL;dr I would like to work in PyCharm on a notebook placed on remote server from my computer using ssh. (How) Can i do that?

**Comment by takluyver** [comment]
I think you'd have to ask PyCharm about that - they have an entirely separate implementation of the notebook interface, and I'm not sure how it works.

**Comment by PiotrowskiD** [comment]
Ok, thank you very much for a quick reply.

**Comment by waqas-shahid** [comment]
@PiotrowskiD did you find out a solution, I am facing the same issue.

**Comment by renegademonkey** [comment]
@PiotrowskiD - I'm also really interested in this. I have successfully set up a regular remote python interpreter via ssh to run python scripts, but would be interested in getting jupyter to work from PyCharm as well. 

Has anyone had any luck?

**Comment by PiotrowskiD** [comment]
Unfortunately no, although I gave up really quickly so maybe you still have a chance

**Comment by renegademonkey** [comment]
​haha. thanks for the update - i'll give it another day or two of fiddling
. . .​

**Comment by MrSaints** [comment]
PyCharm 2019 has improved support for Jupyter. Unfortunately, I could not get it to work with JupyterHub as you can't quite configure the path of the server URL. It keeps removing any paths, retaining the hostname. Give it a try, you may have more luck.

**Comment by renegademonkey** [comment]
@MrSaints - the new Jupiter implementation in PyCharm is much, much better and I've finally got it to work (with the help of several ssh tunnels). . .

**Comment by elgalu** [comment]
Has someone figured this out? PyCharm <-> JupyterHub

VSCode has the same issue: https://github.com/microsoft/vscode-python/issues/3954

**Comment by ivanov** [comment]
Hey there, I'm going through old issues and it seems to me that it makes sense to close this one.

Thanks everyone and happy hacking! :bowtie:
